sympy/stats/rv.py

```diff
@@ -1041,9 +1041,9 @@ def sample(expr, condition=None, size=(), library='scipy',
         by default is 'scipy'
     numsamples : int
         Number of samples, each with size as ``size``
-    seed : int
-        An integer to be used as seed by the external library.
-        This seed will be used specifically for sampling `expr`.
+    seed :
+        An object (like int, numpy.random.RandomState, numpy.random.default_rng)
```

Comment:
```suggestion
        An object (like int, numpy.random.Generator)
```

Since our code uses `np.random.default_rng(seed=seed)` we can't actually use a RandomState(). To enable that we would have to check for the type via `isinstance(seed, np.random.RandomState)` and set the rng accordingly. But as of now, only an int and a Generator should work:

```
np.random.default_rng(seed=np.random.RandomState())
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "_generator.pyx", line 4360, in numpy.random._generator.default_rng
  File "_pcg64.pyx", line 109, in numpy.random._pcg64.PCG64.__init__
  File "bit_generator.pyx", line 526, in numpy.random.bit_generator.BitGenerator.__init__
  File "bit_generator.pyx", line 308, in numpy.random.bit_generator.SeedSequence.__init__
TypeError: SeedSequence expects int or sequence of ints for entropy not RandomState(MT19937)
```

Reply:
When the library is `scipy`, then any one of `numpy.random.RandomState` can be passed as `seed`. See,
```python
In [1]: from sympy.stats import *

In [2]: import numpy

In [3]: rand_state = numpy.random.RandomState(0)

In [4]: N = Normal('N', 0, 1)

In [5]: itr = sample(N, numsamples=10, library='scipy', seed=rand_state)
/home/czgdp1807ssd/sympy_project/sympy/sympy/stats/rv.py:1099: UserWarning: 
The return type of sample has been changed to return an iterator
object since version 1.7. For more information see
https://github.com/sympy/sympy/issues/19061
  warnings.warn(filldedent(message))

In [6]: list(itr)
Out[6]: 
[1.764052345967664, 0.4001572083672233, 0.9787379841057392, 2.240893199201458,
 1.8675579901499675, -0.977277879876411, 0.9500884175255894, -0.15135720829769
79, -0.10321885179355784, 0.41059850193837233]
```
Sampling code for all the supported libraries is independent of each other. That is why, I didn't mention the accepted types of `seed` in the type list (i.e., after the `:`) and mentioned some example types in the description.

Reply:
My bad, I didn't consider this to be a summary list of dependent input types. Should this then be mentioned in the docstring too, so as to not confuse any users? E.g.:

> Allowed types by library:
> - SciPy: int, RandomState, Generator
> - numpy: int, Generator
> - pymc3: int

Also, I would still change `np.random.default_rng` to `np.random.Generator` since one can also pass a different bit generator than PCG64 if they so desire :)

Reply:
Sure. I would push the changes by tomorrow if for accepting `Generator` as `seed` for `numpy` a bunch of `if` checks would be needed to not create a RNG when the `seed` is not an `int`. Well but since, `scipy` is good enough and accepts RNGs of diverse types, may be doing changes for `numpy` will be overwhelming and unnecessary. What do you say?
Otherwise we can just do the following in the docs,

```
Allowed types by library:

- SciPy: int, RandomState, Generator
- numpy: int
- pymc3: int

```

Reply:
I think we should bring the generator to numpy eventually, it isn't that much extra code anyway (i'm assuming two `isinstance` calls would do the job). Stalling this PR until then seems most reasonable to me. I think I can take care of the boilerplate for numpy in the coming week anyway, if you want me to.

Reply:
I'll push the changes shortly for allowing different types of generators for `numpy`.

Reply:
Perfect, thanks for the changes!